Source_Code/Server/ClueLessServer.java

```diff
@@ -286,21 +279,32 @@ public void processAction(ActionRequest actionRequest)
         {
             processConnectRequest((ConnectRequest) actionRequest);
         }
-        if(actionRequest instanceof GameStartRequest)
-        {
-            processGameStartRequest((GameStartRequest) actionRequest);
-        }
-
-        if(actionRequest instanceof MoveRequest)
+        for(Player p : PlayerList)  // Only process non-connection requests from players
         {
-           processMoveRequest((MoveRequest) actionRequest);
-        }
-
-        if(actionRequest instanceof SuggestRequest)
-        {
-            processSuggestRequest((SuggestRequest) actionRequest);
+            if(actionRequest.UniqueID == p.ClientID)
+            {
+                if(actionRequest instanceof GameStartRequest)
+                {
+                    processGameStartRequest((GameStartRequest) actionRequest);
+                }
+                if(PlayerList.get( CurrentPlayerIndex ).PlayerName.equals(p.PlayerName))  // Only the player who's turn it is can do these
```

Comment:
Also need to check if they are "out" which is tracked in the PlayerACtive attribute